### PR TITLE
BLD: update pip requirements, manifest, and general setup things

### LINKED ISSUE #284: pip requirements wrong
* We generally just install with conda, so this isn't critical
* However, the 3-line `requirements.txt` is _very_ out of date with respect to the conda recipe
https://github.com/pcdshub/hutch-python/blob/master/requirements.txt
https://github.com/pcdshub/hutch-python/blob/master/conda-recipe/meta.yaml

Handy pcdsutils requirements comparator:
```
$ requirements-compare hutch-python
--- requirements.txt: host/build/run ---
Missing in pip:
- archapp
- bluesky
- cookiecutter
- elog
- happi
- jinja2
- lightpath
- matplotlib
- nabs
- pcdscalc
- pcdsdaq
- pcdsdevices
- pcdsutils
- psdm_qs_cli
- pyfiglet
- pyqt5
- requests
- simplejson

--- dev-requirements.txt: test ---
Missing in conda:
- doctr
- flake8
- nbsphinx
- pytest
- pytest-timeout
- sphinx
- sphinx_rtd_theme

--- docs-requirements.txt: docs ---
```

### PR DESCRIPTION
**Pull request by klauer**
Closes #284 - for the most part

My quick test indicated _at least_ some work is needed on lightpath on PyPI at the moment. (https://github.com/pcdshub/lightpath/issues/124)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+psdm_qs_cli >=0.3.1
+pyfiglet
+pyqt5
 pyyaml
-coloredlogs
+requests
+simplejson
+
+# Some are unavailable on pypi:
+# elog
```
Comment (ZLLentz) [inline requirements.txt:23]: Do we need to make this an optional dependency?

**Review by ZLLentz — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (klauer) [reply requirements.txt:23]: I think it'd be not so difficult to do. Would you be OK with me taking a stab at that in this PR?

Comment (klauer) [reply requirements.txt:23]: Scratch that. I think it deserves a follow-up PR.

**Comment by klauer** [comment]
What am I doing wrong here? Broke the conda build with the following:
```
pkg_resources.DistributionNotFound: The 'pcdsdevices>=3.1.0' distribution was not found and is required by hutch-python
```

**Comment by ZLLentz** [comment]
There is something wrong with the packaging of pcdsdevices in the most recent tag:
```
In [1]: import pcdsdevices

In [2]: pcdsdevices.__version__
Out[2]: '0+unknown'
```

**Comment by klauer** [comment]
Right, of course, a pypi release wasn't enough just yet. We need the conda-forge release of pcdsdevices as well to have this PR's CI pass.

**Comment by klauer** [comment]
Packaging h-e-double-hockey-sticks: 

```
pkg_resources.DistributionNotFound: The 'versioneer' distribution was not found and is required by qtpyinheritance, pcdscalc
```

**Comment by klauer** [comment]
Removed versioneer from those package's requirements. This PR is stalled again

**Comment by ZLLentz** [comment]
You're finding all sorts of time bombs in these old project setups

**Comment by klauer** [comment]
New versions of offending libraries are on conda-forge. What failures await us now?

**Comment by ZLLentz** [comment]
pip can't find nabs on pypi
otherwise everything passes here :tada:

**Comment by klauer** [comment]
👏

**Pull request merged**